src/telemetry/mod.rs

```diff
@@ -16,7 +16,7 @@ pub fn new(
     metrics_retain_idle: Duration,
     taps: &Arc<Mutex<tap::Taps>>,
 ) -> (Sensors, metrics::Serve) {
-    let (metrics_record, metrics_serve) = metrics::new(process, metrics_retain_idle);
-    let s = Sensors::new(metrics_record, taps);
+    let (metrics_record, tls_config_reload, metrics_serve) = metrics::new(process, metrics_retain_idle);
+    let s = Sensors::new(metrics_record, tls_config_reload, taps);
```

Comment:
TIOLI: I wonder if it's really necessary for `sensors` to own the `TlsConfigReload`? It seems like it's only used once, to get a reference to the `sensors`' `TlsConfigReload` field and clone it, and then the sensors no longer need it. Maybe `telemetry::new` ought to just return `(Sensors, Arc<TlsConfigReload>, metrics::Serve)` instead?

Reply:
this will all change. this seemed like a fine interim state that makes the review small